If WebJar version is a "-snapshot", overwrite existing files when extracting

mkurz opened the pull request:
In general, Snapshots usually indicate things are still wip and changing (e.g., that's why you can overwrite artifacts in the [sonatype snapshot repository](https://s01.oss.sonatype.org/content/repositories/snapshots/)); therefore, existing files should get overwritten when extracting a webjar which obviously is a -snapshot release.

This change alone makes total sense on its own and will not break things, IMHO.

However, the real reason we need this behavior is because it helps us (Play) to fix:
- https://github.com/playframework/playframework/issues/12484
- https://github.com/orgs/playframework/discussions/12636

While developing a Play application and editing assets (css, js, etc.) in a sub-project, the sub-project's version usually is a -snapshot, so when a user edits an asset and refreshes the browser, the updated asset (which is internally a webjar) should be served. Currently, this is broken because files will not get overwritten. (I did test this locally with Play apps, and it works with this PR.)

@jamesward If you give thumbs up for this PR, would be nice if you could cut a new release also, thanks!

jamesward commented:
This seems good to me. The only issue is that for some reason the behavior is different on Windows. :(

mkurz commented:
> The only issue is that for some reason the behavior is different on Windows. :(

Fixed, this was just a new line quirk in the test I added.

BTW. In Play 2.8 this was working because it was using a very outdated webjars-locator-core version that did not have
- #28

merged yet.
With this PR you switched from "do not overwrite a file when it is up to date" to "do not overwrite a file when it exists". You can see that also in the tests you changed: `extractWebJarShouldNotExtractWhenWhenUpToDate` got removed and `extractWebJarShouldNotExtractWhenFileExists` got added. ([diff](https://github.com/webjars/webjars-locator-core/pull/28/files#diff-f55f42723db1181579a4857ddbc81893a416e8a4d249b077551bf07494e013c5R58)).
So, in Play 2.8 when a user edited a file, it was always newer then the extracted one and so it was never "up to date" and the file got overwritten.

Thanks!

jamesward commented:
This is being released as `0.59`

mkurz commented:
You rock!

jamesward commented:
No, Matthias. You totally rock! :)
